### ISSUE TITLE
Code optimization cannot be disabled from make command line

### ISSUE THREAD

**Issue by raphaelt-nvidia** [Azure/sonic-buildimage#6413]
**Description**

Including "SONIC_PROFILING_ON=y" in the make command line does nothing.  The code is optimized anyway.

**Steps to reproduce the issue:**

For example,

make SONIC_BUILD_JOBS=12 INSTALL_DEBUG_TOOLS=y SONIC_PROFILING_ON=y target/debs/stretch/swss_1.0.0_amd64.deb

**Describe the results you received:**

Output produced during the make shows

"SONIC_PROFILING_ON"              : ""

Running in gdb, it is clear that optimization is on.

**Describe the results you expected:**

"SONIC_PROFILING_ON=y" should turn optimization off.

**Additional information you deem important:**

Optimization can be turned off if this line is uncommented in sonic-buildimage/rules/config:

SONIC_PROFILING_ON = y

The reason it does not work from the command line is that where Makefile.work invokes slave.mk in the paragraph that begins thus:
 
SONIC_BUILD_INSTRUCTION :=  make \
                           -f slave.mk \
                           BLDENV=$(BLDENV) \
 
SONIC_PROFILING_ON is missing from the list of options passed to slave.mk.

Fixing this should include checking for other options missing from that list.

**Comment by anshuv-mfst** [comment]
@lguohan could you please tag this issue. 

@Dror - Could you please provide workaround mentioned in the triage meeting, thanks.

**Comment by dprital** [comment]
WorkAround: 

Can be set by uncomment this line in sonic-buildimage/rules/config:
SONIC_PROFILING_ON = y

**Comment by lguohan** [comment]
is this similar to #5982

**Comment by raphaelt-nvidia** [comment]
No.  In the issue I opened, the compile succeeds, but optimization is enabled in spite of the command line option.

Raphael Tryster
Staff Engineer, Software

**Comment by raphaelt-nvidia** [comment]
It is related to https://github.com/Azure/sonic-buildimage/issues/5982 in that 5982 describes the workaround for this bug, which doesn't work.  That is, if I want to turn off optimization, I edit rules/config to set SONIC_PROFILING_ON = y, but then the compile fails.  This forces a very roundabout way of working: compile the whole project with optimization o as usual, then set SONIC_PROFILING_ON = y and compile just a few modules like swss or syncd which don't fail when this flag is set.
